src/model/Spectrum.cpp

```diff
@@ -408,24 +441,65 @@ bool Spectrum::parsePwizSpecInfo(
   first_scan_ = firstScan;
   last_scan_ = lastScan;
 
-  // get peaks
+  // Get the retention time
+  //  MS_retention_time = 1000894,
+  /// retention time: A time interval from the start of chromatography when an analyte exits a chromatographic column.
+  if (pwiz_spectrum->hasCVParam(pwiz::msdata::MS_retention_time)) {   
+    retention_time_ = pwiz_spectrum->cvParam(pwiz::msdata::MS_retention_time).valueAs<double>();
+  // MS_scan_start_time = 1000016,
+  /// scan start time: The time that an analyzer started a scan, relative to the start of the MS run.    
+  } else if (!pwiz_spectrum->scanList.empty() && pwiz_spectrum->scanList.scans[0].hasCVParam(pwiz::msdata::MS_scan_start_time)) {   
+    retention_time_ = pwiz_spectrum->scanList.scans[0].cvParam(pwiz::msdata::MS_scan_start_time).valueAs<double>();
+  }
+  // The following does not work.
+  // } else if (pwiz_spectrum->scanList.scans[0].hasCVParam(pwiz::msdata::MS_retention_time)) {   
+  //   retention_time_ = pwiz_spectrum->scanList.scans[0].cvParam(pwiz::msdata::MS_retention_time).valueAs<double>();
+  // }
+ // get peaks
   int num_peaks = pwiz_spectrum->defaultArrayLength;
   vector<double> mzs = pwiz_spectrum->getMZArray()->data;
   vector<double> intensities = pwiz_spectrum->getIntensityArray()->data;
-  for(int peak_idx = 0; peak_idx < num_peaks; peak_idx++){
-    addPeak(intensities[peak_idx], mzs[peak_idx]);
-  }
+
+  // if (dia_mode || Params::GetBool("skip-preprocessing") ) {
+    for (int peak_idx = 0; peak_idx < num_peaks; peak_idx++) {
+      addPeak(intensities[peak_idx], mzs[peak_idx]);
+    }
+  // } else { //  This should run only for tide-search.
```

Comment:
Are these commented lines necessary to keep?

Reply:
I think we will remove it.